canvas/canvas_course_usage.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import time
+import pandas as pd
+from json.decoder import JSONDecodeError
+from requests_futures.sessions import FuturesSession
+from concurrent.futures import as_completed
+import json
+logger = logging.getLogger(__name__)
+
+
+class CanvasCourseUsage:
+    def __init__(self, canvas_url, canvas_token, retry_attempts, course_ids):
+        self.canvas_url = canvas_url,
+        self.canvas_token = canvas_token,
+        self.course_ids = course_ids
+        self.retry_attempts = retry_attempts
+        self.canvas_usage_courses = []
+        self.course_retry_list = []
+        self.retry_count = 0
+
+    def parsing_canvas_course_usage_data(self, response):
+        logger.info("parsing_canvas_course_usage_data Call")
+        new_dic = {}
+        if response is None:
+            logger.info(f"For Canvas Course usage response is None ")
+            return
```

Comment:
Nitpick: Can you return `None` when you do this? Seems a little clearer to me. I'm not insisting on you making the change.

Reply:
I never thought much about this before. In the [PEP-8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) it says "Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should.". In this function it looks okay but I also like the explicit `return None` and have been trying to use it more and remember to check more for it. Either `return None` and `return` both still return None. 

FYI: If you were type hinting this putting `-> None: ` would make it clear that this only returns None.